### PR TITLE
Add Dssp

### PR DESCRIPTION
**Pull request by berkeucar**
I added DSSP (https://swift.cmbi.umcn.nl/gv/dssp/index.html) with slightly changing the script from https://anaconda.org/salilab/dssp. 

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

### General instructions

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

### Instructions for avoiding API, ABI, and CLI breakage issues
Conda is able to record and lock (a.k.a. pin) dependency versions used at build time of other recipes.
This way, one can avoid that expectations of a downstream recipe with regards to API, ABI, or CLI are violated by later changes in the recipe.
If not already present in the meta.yaml, make sure to specify `run_exports` (see [here](https://bioconda.github.io/contributor/linting.html#missing-run-exports) for the rationale and comprehensive explanation).
Add a `run_exports` section like this:

```yaml
build:
  run_exports:
    - ...

```

with `...` being one of:

| Case                             | run_exports statement                                               |
| -------------------------------- | ------------------------------------------------------------------- |
| semantic versioning              | `{{ pin_subpackage("myrecipe", max_pin="x") }}`     |
| semantic versioning (0.x.x)      | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}`   |
| known breakage in minor versions | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| known breakage in patch versions | `{{ pin_subpackage("myrecipe", max_pin="x.x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| calendar versioning              | `{{ pin_subpackage("myrecipe", max_pin=None) }}`    |

while replacing `"myrecipe"` with either `name` if a `name|lower` variable is defined in your recipe or with the lowercase name of the package in quotes.

### Bot commands for PR management

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

Note that the <code>@BiocondaBot please merge</code> command is now depreciated. Please just squash and merge instead.

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by berkeucar** [comment]
@BiocondaBot please add label

**Comment by berkeucar** [comment]
@BiocondaBot please fetch artifacts

File: recipes/dssp/build.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+sed -i -e 's/^CXX/#CXX/' makefile
+
+# clang++ won't find std::tuple unless in C++11 mode;
+# don't compile with anything newer than C++11 since we use the removed
+# std::unary_function
+if [ `uname -s` = "Darwin" ]; then
+  CFLAGS="-std=c++11 -Wno-enum-constexpr-conversion ${CFLAGS}"
+fi
+
+make -j4 install
```
Comment (martin-g) [inline recipes/dssp/build.sh:23]:
```suggestion
make -j${CPU_COUNT} install
```

File: recipes/dssp/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,52 @@
+{% set version = "3.0.0" %}
+{% set name = "dssp" %}
+# This file, the patch file and build file were taken from conda salilab channel's recipes - please see https://github.com/salilab/conda-recipes for more.
+package:
+  name: {{ name }}
+  version: {{ version }}
+
+source:
+  url: https://salilab.org/dssp/dssp-3.0.0.tgz
```
Comment (martin-g) [inline recipes/dssp/meta.yaml:9]:
```suggestion
  url: https://salilab.org/dssp/dssp-{{ version }}.tgz
```

File: recipes/dssp/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+build:
+  number: 0
+  run_exports:
+    - {{ pin_subpackage('dssp', max_pin="x.x") }}
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - boost-cpp
```
Comment (martin-g) [inline recipes/dssp/meta.yaml:23]: Only 'make' and the compilers should be in 'build'. The libraries should be in 'host'

File: recipes/dssp/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    - zlib
+    - bzip2
+    - make
+  run:
+    - {{ pin_compatible('boost-cpp') }}
+
+about:
+  home: http://swift.cmbi.ru.nl/gv/dssp/
+  summary: 'Secondary structure assignment'
+  license: Boost
```
Comment (martin-g) [inline recipes/dssp/meta.yaml:33]: license_file ?

**Comment by martin-g** [comment]
There is a recipe in conda-forge for it already: https://anaconda.org/conda-forge/dssp

**Pull request closed without merging**